common-libs/common/src/services/DidDocumentService/index.ts

```diff
@@ -1,28 +1,31 @@
 import KeysService from '../KeysService'
 import JoloDidDocument from './JoloDidDocument'
 import ElemDidDocument from './ElemDidDocument'
+import ElemAnchoredDidDocument from './ElemAnchoredDidDocument'
 import { parse } from 'did-resolver'
 import { LocalKeyVault } from './LocalKeyVault'
+import { DidResolver } from '../../shared/DidResolver'
 
 export { default as ElemDidDocument } from './ElemDidDocument'
 export { KeyVault } from './KeyVault'
 export { LocalKeyVault } from './LocalKeyVault'
 
 export default class DidDocumentService {
-  constructor(keysService: KeysService) {
-    const { didMethod } = keysService.decryptSeed()
+  /**
+   * @deprecated use DidDocumentService.createDidDocumentService instead
+   */
+  constructor(keysService: KeysService, didResolver?: DidResolver) {
+    return DidDocumentService.createDidDocumentService(keysService, didResolver)
+  }
 
-    let didDocumentService
-    switch (didMethod) {
-      case 'jolo':
-        didDocumentService = new JoloDidDocument(keysService)
-        break
-      case 'elem':
-        didDocumentService = new ElemDidDocument(new LocalKeyVault(keysService))
-        break
-    }
+  static createDidDocumentService(keysService: KeysService, didResolver?: DidResolver) {
+    const { didMethod } = keysService.decryptSeed()
 
-    return didDocumentService
+    return {
+      jolo: new JoloDidDocument(keysService),
+      elem: new ElemDidDocument(new LocalKeyVault(keysService)),
+      'elem-anchored': new ElemAnchoredDidDocument(keysService, didResolver),
+    }[didMethod]
   }
 
   getMyDid() {
```

Comment:
So all these non-static methods are not needed anymore, because there never were any instances of `DidDocumentService`, and now there are no variables with that type as well.